src/GraphCoarsen.cpp

```diff
@@ -508,7 +514,14 @@ void GraphCoarsen::BuildPEdges(
                                  nedges, total_num_traces + bubble_counter);
     Pedges.Swap(newPedges);
 
-    CoarseM_ = mbuilder.GetCoarseM(M_v, Pedges, face_cdof);
+    auto coef_mbuilder_ptr = dynamic_cast<CoefficientMBuilder*>(&mbuilder);
+    if (coef_mbuilder_ptr)
+    {
+        coef_mbuilder_ptr->BuildComponents(M_v, Pedges, face_cdof);
+    }
+
+    // TODO: only compute CoarseM_ when solver is not hybridization
+    CoarseM_ = mbuilder.GetAssembledM();
```

Comment:
Shorter version: I think you can remove lines 523-524 now, and later on we should figure out why.

Reply:
Good catch! Yes, I think this is because the role of the M matrix is replaced by `mbuilder` now, and like you said the M matrix will be constructed using `mbuilder->GetAssembledM()` (this is also called in `MixedMatrix::getWeight()`).